### ISSUE TITLE
Type error during Semantic Error resporting

### ISSUE THREAD

**Issue by christian-herber** [CZ-NIC/yangson#95]
When validating data, I am sometimes faced with the situation that the expression passed to the SematicError actually raises another error:

```
  File "...\site-packages\yangson\schemanode.py", line 965, in _validate
    raise SemanticError(inst.json_pointer(expand_keys=True), "instance-required")
  File "...\site-packages\yangson\instance.py", line 199, in json_pointer
    res.insert(0, inst.schema_node.name + '=' + inst.value)
TypeError: can only concatenate str (not "int") to str
```

This could be an issue in other functionalities also, but here it is a bit annoying, because the Semantic Error cannot be reported. In this specific case, I have a LeafList with LeafRefs to an Integer node.

My proposal would be to change
``` C
res.insert(0, inst.schema_node.name + '=' + inst.value)
```
to
``` 
res.insert(0, '{}={}'.format(inst.schema_node.name, inst.value))
```

**Comment by llhotka** [comment]
This is related to #91 and has to do with the new `expand_keys` argument of the `json_pointer` method.

My proposal is to remove this argument, but also modify all instance-related exceptions (such as `SematicError`) to have the offending instance as its member, instead of `path` (JSON pointer). An exception handler could then inspect the instance directly.

@kwatsen @HRogge any thoughts on this?

**Comment by llhotka** [comment]
Fixed by PR #97 (merged in 1.4.0)
